RELEASE.md

```diff
@@ -0,0 +1,100 @@
+
+Note that `CHANGELOG.md` should only document changes relevant to users of Prometheus, including external API changes, performance improvements, and new features. Do not document changes of internal interfaces, code refactorings and clean-ups, changes to the build process, etc. People interested in these are asked to refer to the git history.
+
+Entries in the `CHANGELOG.md` are meant to be in this order:
+
+* `[CHANGE]`
+* `[FEATURE]`
+* `[ENHANCEMENT]`
+* `[BUGFIX]`
+
+### Draft the new release
+
+Tag the new release with a tag named `v<major>.<minor>.<patch>`, e.g. `v2.1.3`. Note the `v` prefix.
+
+You can do the tagging on the commandline:
+
+```bash
+$ tag=$(< VERSION)
+$ git tag -s "v${tag}" -m "v${tag}"
+$ git push --tags
+```
+
+Signing a tag with a GPG key is appreciated, but in case you can't add a GPG key to your Github account using the following [procedure](https://help.github.com/articles/generating-a-gpg-key/), you can replace the `-s` flag by `-a` flag of the `git tag` command to only annotate the tag without signing.
+
+Once a tag is created, the release process through CircleCI will be triggered for this tag.
+You must create a Github Release using the UI for this tag, as otherwise CircleCI will not be able to upload tarballs for this tag. __Also, you must create the Github Release using a Github user that has granted access rights to CircleCI.__ If you did not or cannot grant those rights to your personal account, you can log in as `prombot` in an anonymous browser tab. (This will, however, prevent verified releases signed with your GPG key. For verified releases, the signing identity must be the same as the one creating the release.)
+
+Go to the releases page of the project, click on the _Draft a new release_ button and select the tag you just pushed. The title of the release is formatted `x.y.z / YYYY-MM-DD`. Add the relevant part of `CHANGELOG.md` as description. Click _Save draft_ rather than _Publish release_ at this time. (This will prevent the release being visible before it has got the binaries attached to it.)
+
+You can also create the tag and the Github release in one go through the Github UI by going to the releases page and then click on the `Draft a new release` button and enter your tag version.
+
+Now all you can do is to wait for tarballs to be uploaded to the Github release and Docker images to be pushed to the Docker Hub and Quay.io. Once that has happened, click _Publish release_, which will make the release publicly visible and create a GitHub notification.
+
+### Wrapping up
+
+If the release has happened in the latest release branch, merge the changes into master.
+
+To update the docs, a PR needs to be created to `prometheus/docs`. See [this PR](https://github.com/prometheus/docs/pull/952/files) for inspiration.
+
+Once the binaries have been uploaded, announce the release on `prometheus-users@googlegroups.com`. Start the subject with `[ANN]`. Check out previous announcement mails for inspiration.
```

Comment:
Note that this has to be changed once we have the `prometheus-announce@` list in action (i.e. don't use `[ANN]` as that's now already in the prefix of the list, and obviously use `-announce` instead of `-users`).

BTW: What's currently happening to `prometheus-announce@`? We either have to announce that everybody interested in announcements needs to subscribe there, or we have to settle on mirroring announcements to `prometheus-users@`.

Reply:
I think I missed most of the discussion on IRC, for the 2.5 release I posted announcements on all 3 mailing lists, should we continue to do this, or only the announcement list?

Reply:
That's a good question. If I haven't missed any part of the discussion, it just ran dry and stopped before coming to a conclusion. As `-announce` was not yet, well, announced, I would merge this PR as-is and update once we have a result.

Reply:
Agreed. Let's move forward and adapt as we come to a conclusion. :+1: